src/snapshot.ts

```diff
@@ -0,0 +1,346 @@
+import {
+  bufferToFile,
+  fileToBuffer,
+  mySessionItems,
+  mySessionMeat,
+  print,
+  sellPrice,
+  toInt,
+  toItem,
+} from "kolmafia";
+import { $item, $items, getFoldGroup, getSaleValue, property, set, sumNumbers } from "libram";
+import { fmt, HIGHLIGHT, resetDailyPreference } from "./lib";
+
+/**
+ * Return a mapping of the session items, mapping foldable items to a single of their forms
+ * @returns the item session results, with foldables mapped to a single of their folding forms
+ */
+function mySessionItemsWrapper(): Map<Item, number> {
+  const foldableMapping = (item: Item): [Item, Item][] =>
+    getFoldGroup(item).map((target: Item) => [target, item]);
+
+  const foldables = new Map<Item, Item>([
+    ...foldableMapping($item`liar's pants`),
+    ...foldableMapping($item`ice pick`),
+    ...foldableMapping($item`Spooky Putty sheet`),
+    [$item`Spooky Putty monster`, $item`Spooky Putty sheet`],
+    ...foldableMapping($item`stinky cheese sword`),
+    ...foldableMapping($item`naughty paper shuriken`),
+    ...foldableMapping($item`Loathing Legion knife`),
+    ...foldableMapping($item`deceased crimbo tree`),
+    ...foldableMapping($item`makeshift turban`),
+    ...foldableMapping($item`turtle wax shield`),
+    ...foldableMapping($item`metallic foil bow`),
+    ...foldableMapping($item`ironic moustache`),
+    ...foldableMapping($item`bugged balaclava`),
+    ...foldableMapping($item`toggle switch (Bartend)`),
+    ...foldableMapping($item`mushroom cap`),
+    [$item`empty Rain-Doh can`, $item`can of Rain-Doh`],
+  ]);
+
+  const inventory = new Map<Item, number>();
+  for (const [itemStr, quantity] of Object.entries(mySessionItems())) {
+    const item = toItem(itemStr);
+    const foldableItem = foldables.get(item) ?? item;
+    const alreadyInventory = inventory.get(item);
+    if (alreadyInventory !== undefined) {
+      inventory.set(foldableItem, quantity + alreadyInventory);
+    } else {
+      inventory.set(foldableItem, quantity);
+    }
+  }
+  return inventory;
+}
+
+/**
+ * Performa a binary element-wise operation on two inventories
+ * @param a The LHS inventory to perform the operation on
+ * @param b The RHS inventory to perform the operation on
+ * @param op a function to compute between the sets
+ * @param transitive if true use the value of b for any items not in a. if false, ignore values not in a
+ * @returns a new map representing the combined inventories
+ */
+function inventoryOperation(
```

Comment:
I feel like this function could benefit from generic typing, but this is based on vibes alone